docs/proposals/scheduling/20240115-elastic-quota-support-job-level-oversold-management.md

```diff
@@ -0,0 +1,295 @@
+    - [References](#references)
+
+<!-- /TOC -->
+
+## Summary
+This proposal provides job level elastic quota oversold mechanism for the scheduler, including resource assign and preemption.
+
+## Motivation
+In a large scale shared-cluster, some quotas may be very busy, some quotas may be idle. If the idle resources
+can be shared in some way, the resource utilization rate of the cluster will be significantly improved. Moreover, even in
+one quota, there are production jobs and develop jobs, user may expect production jobs can preempt develop jobs
+when resource is busy.
+
+If the pods belong to a job , we should also consider resource assign\preemption by job level.
+
+### Goals
+
+1. Define API to announce guaranteed and oversold job/pod. This proposal we use non-preemptible and preemptible instead of
+   guaranteed and oversold.
+
+2. Enhance the elastic quota plugin to achieve job level elastic quota oversold management.
+
+### Non-goals/Future work
+
+## Proposal
+
+### Key Concept\User Stories
+In our scenario, a user may create several quotas. And operators will select and bind some nodes to a concrete quota,
+normally job can only run in quota's binding nodes(Through label\affinity method). However, some quotas may be busy and some
+quotas may be idle, so the user want to submit some preemptible job belongs to busy-quota and running on idle-quota's binding nodes.
+When idle-quota turns to busy-quota, the user want the preemptible job should be preempted by idle-quota's non-preemptible job.
+
+However, node and quota binding should not have affect to preemption, so the doc below only introduce the node and quota binding
+in user-view, the concrete preemption implementation has no connection with quota node binding concept.
+
+We use label `pod-group.scheduling.sigs.k8s.io` on pod to describe server pod belongs to one job. If job is preemptible,
+please fill job's pods with label `quota.scheduling.koordinator.sh/preemptible=true`, if job is non-preemptible,
+please fill job's pods with label `quota.scheduling.koordinator.sh/preemptible=false`. Typically, we use pod.spec.PreemptionPolicy
+to judge whether a pod can preempt preemptible pods. If pod's policy is PreemptLowerPriority, it can trigger preemption. 
+For hot-update scenario, the cluster exist pods' pod.spec.PreemptionPolicy may all PreemptLowerPriority, but we only want 
+some of the pods to trigger preemption. So, we also use pod's label to declare whether the pod can trigger preemption or not.
+If job can trigger preemption, please fill job's pods with label `quota.scheduling.koordinator.sh/can-preempt=true`. If 
+job can not trigger preemption, please fill job's pods with label `quota.scheduling.koordinator.sh/can-preempt=false`. Priority-class 
+is also used for preemption, for example high priority-class pods can preempt low-priority-class pod. 
+
+1. Each quota group declares its own "min" and "max". The semantics of "min" is the quota group's non-preemptible resources.
```

Comment:
These concepts have been introduced in previous PRs(#1668, #1755), but there are indeed no detailed Proposals. The motivation and background of these conceptual changes are hidden in the PRs. It is indeed necessary to take this opportunity to reintroduce the changes in min/max.

Reply:
done